client/src/hooks.ts

```diff
@@ -100,3 +100,24 @@ export function useOnlineStatus(): { isOnline: boolean; isOffline: boolean } {
 
   return { isOnline, isOffline };
 }
+
+/**
+ * Scrolls after a client-side navigation.
+ *
+ * Limitation: Overrides the scroll position on backward/forward navigation.
+ *
+ * @param deps Values that trigger a client-side navigation when changed.
+ */
+export function useScrollEffect(deps: React.DependencyList) {
+  useEffect(() => {
+    if (window) {
+      const id = window.location.hash.replace("#", "");
+      if (id && document.getElementById(id)) {
+        window.location.href = window.location.hash;
+      } else {
+        window.scrollTo(0, 0);
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, deps);
```

Comment:
Why did you need to disable the eslint rule here? Is it because eslint does not know that `deps` is an array?

Reply:
Indeed ESLint cannot check that all dependencies inside the effect are also a dep, as deps is passed in from the caller. 

It seems to be a limitation of that ESLint rule, because this hook doesn't actually use any React dependency.

Reply:
If it does not have any dependencies, could we use `[]`? Or can it potentially have a dependency array? If it should run every time, then we can leave off the `[]` part entirely.